homeassistant/components/homeassistant_hardware/silabs_multiprotocol_addon.py

```diff
@@ -358,16 +447,19 @@ async def async_step_install_addon(
             self.install_task = None
             _LOGGER.error(err)
             return self.async_show_progress_done(next_step_id="install_failed")
-
-        self.install_task = None
+        finally:
+            self.install_task = None
```

Comment:
Either remove the finally clause, ore remove `self.install_task = None` from the exception handler just above